pkg/sql/parser/lexer.go

```diff
@@ -54,6 +54,14 @@ func (l *lexer) init(
 	l.nakedSerialType = nakedSerialType
 }
 
+// cleanup is used to avoid holding on to memory unnecessarily (for the cases
+// where we reuse a scanner).
+func (l *lexer) cleanup() {
+	l.tokens = nil
```

Comment:
Does it make sense to do something like `if cap(l.tokens) > 1000 { l.tokens = nil } else { l.tokens = l.tokens[:0] }` to avoid reallocating on repeated statements?

Reply:
The lexer doesn't allocate data, this is just an input to it. The slice comes from the scanner, and we initialize that with a preallocated buffer embedded in the parser. We could reuse that if it's reallocated, but if we're ok with holding on to a bigger buffer forever we should just increase the size of the embedded buffer.